### ISSUE TITLE
Invalid structure while fetching an object which has a field of same name

### ISSUE THREAD

**Issue by ghost** [forcedotcom/cli#579]
### Summary

I have created an object named Branch__c, which has the same name field Branch__c and fetching it with sfdx.

### Steps To Reproduce:

1. Create an object Branch__c.
2. Create a field Branch__c.
3. Try to fetch the object with sfdx command: **sfdx force:source:retrieve -m "CustomObject:Branch__c"**

### Expected result

A valid folder and file structure should have been fetched.
**Branch__c** Folder
--->Fields Folder
--->**Branch__c.object-meta.xml** object meta file

### Actual result

Improper folder and file structure was created,
**Branch__c** Folder
--->Fields Folder
--->**Branch__c.field-meta.xml**
--->**Branch__c.object-meta.xml** object meta-file

![image](https://user-images.githubusercontent.com/7561782/91289088-78312080-e7af-11ea-8a36-26b096ff2f17.png)

Also, while uploading this object again, it gives the error:
**An object 'objects.Branch__c' of type CustomField was named in package.xml, but was not found in zipped directory**

### Additional information

sfdx-cli/7.70.1-3d8a6229fc win32-x64 node-v12.18.3
PS C:\Projects\HSdev3-git> sfdx plugins --core
@oclif/plugin-autocomplete 0.2.0
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.0
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.0 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/lwc-dev-server 2.5.1
└─ @oclif/plugin-update 1.3.10
@salesforce/sfdx-diff 0.0.6
@salesforce/sfdx-trust 3.2.1 (core)
analytics 1.12.0 (core)
generator 1.1.3 (core)
salesforce-alm 49.3.1-4 (mpd)
salesforcedx 49.6.1 (core)
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
├─ custom-metadata 1.0.8 (core)
└─ salesforcedx-templates 49.2.0 (core)
sfdmu 3.5.10
sfdx-cli 7.70.1 (core)

Latest stable Windows 10.

**Comment by maaaaarco** [comment]
Yeah confirm, we are getting the same. In our case is a RecordType for the Opportunity object called Opportunity

**Comment by shetzel** [comment]
Per the plugins list (thank you for providing that!) I see you're using the mpd canary release.  Does the problem also reproduce with the core production salesforce-alm plugin?  Just run `sfdx plugins:uninstall salesforce-alm` to get back to the core plugin.  You can reinstall the canary after testing if you prefer it.

**Comment by maaaaarco** [comment]
@shetzel I'm facing same behaviour and I do not use the salesforce-alm:
```
$ sfdx plugins --core
@oclif/plugin-autocomplete 0.2.0
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.0 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/analytics 0.22.0
@salesforce/lwc-dev-server 2.5.1
├─ @oclif/plugin-help 2.2.1
└─ @oclif/plugin-update 1.3.9
@salesforce/sfdx-trust 3.2.1 (core)
analytics 1.12.0 (core)
generator 1.1.3 (core)
salesforcedx 49.7.0 (core)
├─ salesforcedx-templates 49.4.0 (core)
├─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
├─ custom-metadata 1.0.8 (core)
└─ salesforce-alm 49.7.0 (core)
sfdx-cli 7.71.0 (core)
```

**Comment by shetzel** [comment]
@maaaaarco - thank you!

**Comment by ghost** [comment]
@shetzel Well I couldn't reproduce the issue after uninstalling the mpd version. Maybe it is not there for fields but for record types it exists as @maaaaarco  confirmed.

sfdx plugins --core
@oclif/plugin-autocomplete 0.2.0
@oclif/plugin-commands 1.3.0 (core)
@oclif/plugin-help 3.2.0
@oclif/plugin-not-found 1.2.4 (core)
@oclif/plugin-plugins 1.9.0 (core)
@oclif/plugin-warn-if-update-available 1.7.0 (core)
@oclif/plugin-which 1.0.3 (core)
@salesforce/lwc-dev-server 2.5.1
└─ @oclif/plugin-update 1.3.10
@salesforce/sfdx-diff 0.0.6
@salesforce/sfdx-trust 3.2.1 (core)
analytics 1.12.0 (core)
generator 1.1.3 (core)
salesforce-alm 49.3.1-4 (mpd)
salesforcedx 49.7.0 (core)
├─ custom-metadata 1.0.8 (core)
├─ salesforcedx-templates 49.4.0 (core)
└─ @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
sfdmu 3.5.11
sfdx-cli 7.71.0 (core)
